iam.tf

```diff
@@ -0,0 +1,44 @@
+data "aws_iam_policy_document" "this" {
+  # TODO figure out why describetags is needed and if it can be
+  # constrained to the ASG
+  statement {
+    effect    = "Allow"
+    actions   = ["ec2:DescribeTags"]
+    resources = ["*"]
+  }
+
+  statement {
+    actions   = ["secretsmanager:GetSecretValue"]
+    effect    = "Allow"
+    resources = ["arn:aws:secretsmanager:${local.region}:${local.account_id}:secret:${var.scim_secret_name}-*"]
+  }
+}
+
+resource "aws_iam_role_policy" "this" {
+  name   = "${var.name}-policy"
+  policy = data.aws_iam_policy_document.this.json
+  role   = aws_iam_role.this.id
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name               = "${var.name}-role"
```

Comment:
I saw somewhere all these interpolated names were set in the locals. It's really clean especially when you want to allow overrides

Reply:
example? that sounds neat

Reply:
It's what I was doing in the elasticache module, then I've been using `coalesce` as a way to allow overides: https://github.com/rhythmictech/terraform-aws-elasticache/blob/4ffd4879ef68a52949fe51fec675e1cb151a36d9/locals.tf#L8

Reply:
I'mma start doing that more. You'll see it in a PR coming soon.